gentoo/gentoo
# dev-python/jaraco-itertools: bump to 1.8, add doc

Also bump to EAPI=6 and remove a now useless check before running tests.
I also wanted to add [this](https://raw.githubusercontent.com/jaraco/jaraco.itertools/master/CHANGES.rst) as a `changelog` metadata URL, would it be a good idea? Should I use the HTML page? 
```diff
--- jaraco-itertools-1.7.1.ebuild	2016-04-16 22:45:31.898646257 +0200
+++ jaraco-itertools-1.8.ebuild	2016-09-02 23:07:28.443786537 +0200
@@ -1,8 +1,8 @@
-# Copyright 1999-2015 Gentoo Foundation
+# Copyright 1999-2016 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Id$
 
-EAPI=5
+EAPI=6
 
 PYTHON_COMPAT=( python{2_7,3_4,3_5} )
 
@@ -16,13 +16,20 @@
 LICENSE="MIT"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-IUSE="test"
+IUSE="doc test"
 
-RDEPEND="dev-python/six[${PYTHON_USEDEP}]
-dev-python/inflect[${PYTHON_USEDEP}]
-dev-python/more-itertools[${PYTHON_USEDEP}]"
-DEPEND="dev-python/setuptools[${PYTHON_USEDEP}]
+RDEPEND="
+	dev-python/six[${PYTHON_USEDEP}]
+	dev-python/inflect[${PYTHON_USEDEP}]
+	dev-python/more-itertools[${PYTHON_USEDEP}]
+"
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
 	>=dev-python/setuptools_scm-1.9[${PYTHON_USEDEP}]
+	doc? (
+		dev-python/sphinx[${PYTHON_USEDEP}]
+		dev-python/rst-linker[${PYTHON_USEDEP}]
+	)
 	test? (
 		${RDEPEND}
 		>=dev-python/pytest-2.8[${PYTHON_USEDEP}]
@@ -32,13 +39,15 @@
 
 S="${WORKDIR}/${MY_PN}-${PV}"
 
-python_prepare_all() {
-	if use test && has_version "${CATEGORY}/${PN}"; then
-		die "Ensure $PN is not already installed or the test suite will fail"
-	fi
-	distutils-r1_python_prepare_all
+python_compile_all() {
+	use doc && esetup.py build_sphinx
 }
 
 python_test() {
 	PYTHONPATH=. py.test || die "tests failed with ${EPYTHON}"
 }
+
+python_install_all() {
+	use doc && local HTML_DOCS=( "${BUILD_DIR}"/sphinx/html/. )
+	distutils-r1_python_install_all
+}

```
